rust.md

```diff
@@ -0,0 +1,146 @@
+
+The following is a list of rules and best practices for writing Ewasm contracts.
+
+### `lib.rs`, not `main.rs`
+
+As stated earlier, using `main.rs` as a crate root will produce an executable rather than a library.
+
+### Export `main`
+
+Every Ewasm contract is required to export a `main` function as the main entry point for execution. In Rust, simply add the `pub` and `extern "C"` qualifiers to do this:
+```rust
+#[no_mangle]
+pub extern "C" fn main() {
+	// function body goes here
+}
+```
+The `no_mangle` directive should be included as well, to ensure that the compiler will not mangle the `main` identifier and invalidate it.
+
+### Use `finish` and `revert` to end execution
+
+When writing code for a native platform, ending execution properly is usually handled behind the scenes. In Ewasm, there are two EEI functions that signal to end execution:
+
+* [`finish`](https://github.com/ewasm/design/blob/master/eth_interface.md#finish) behaves like EVM's `return`. Simply put, it sets the output data and halts the VM.
+* [`revert`](https://github.com/ewasm/design/blob/master/eth_interface.md#revert) is the exact same thing as EVM's `revert`. It halts the VM and refunds any gas.
+
+### Minimize the use of the standard library
+
+Many constructs in Rust's standard library are not optimized for usage in Ewasm, or are outright unavailable due to dependence on a native syscall interface.
+
+Minimizing the use of non-primitive standard library constructs will improve the size, performance, and gas efficiency of resulting Wasm binaries. Where possible, follow these rules of thumb:
+* Minimize the usage of the heap, in general. Don't use `Box<T>` or `Rc<T>`, or any other smart pointers.
+* Avoid using types with implicitly heap-allocated data. Use arrays instead of `Vec`, and `str` instead of `String`.
+* Borrowing is your friend. Stop using `clone()` and passing non-trivial structs by value, use lifetime parameters and references instead.
+
+### Use the [ewasm-rust-api](https://github.com/ewasm/ewasm-rust-api)
+
+The `ewasm-rust-api` is a convenient library providing safe wrappers around the native EEI functions, in addition to a set of useful EVM-style types.
+
+To use it in your contract, simply add `ewasm_api` as a dependency to `Cargo.toml` and include it in your project.
+
+## 4. Deploying Ewasm contracts
+
+There is one final step that must be done before contract creation: post-processing. 
+
+The Rust compiler, at the moment, does not have fine-grained control over the imports and exports generated other than by their identifiers in the source. As a result, unnecessary exports must be removed and malformed imports corrected.
+
+This can be done using an automated tool such as [wasm-chisel](https://github.com/wasmx/wasm-chisel), or by hand in the `.wast` file produced by disassembling the Wasm binary.
```

Comment:
Could you demonstrate how to run `wasm-chisel` here so these instructions are complete?